android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java

```diff
@@ -0,0 +1,290 @@
+package ti.modules.titanium.ui.widget.tabgroup;
+
+import android.graphics.drawable.Drawable;
+import android.support.design.internal.BottomNavigationItemView;
+import android.support.design.internal.BottomNavigationMenuView;
+import android.support.design.widget.BottomNavigationView;
+import android.view.MenuItem;
+import android.view.View;
+
+import org.appcelerator.kroll.common.Log;
+import org.appcelerator.titanium.TiApplication;
+import org.appcelerator.titanium.TiBaseActivity;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.TiDimension;
+import org.appcelerator.titanium.util.TiColorHelper;
+import org.appcelerator.titanium.util.TiUIHelper;
+import org.appcelerator.titanium.view.TiCompositeLayout;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+
+import ti.modules.titanium.ui.TabGroupProxy;
+import ti.modules.titanium.ui.TabProxy;
+
+/**
+ * TabGroup implementation using BottomNavigationView as a controller.
+ */
+public class TiUIBottomNavigationTabGroup extends TiUIAbstractTabGroup implements MenuItem.OnMenuItemClickListener
+{
+	// region private fields
+	private int mBottomNavigationHeightValue;
+	// BottomNavigationView lacks anything similar to onTabUnselected method of TabLayout.OnTabSelectedListener.
+	// We track the previously selected item index manually to mimic the behavior in order to keep parity across styles.
+	private int currentlySelectedIndex = -1;
+	private BottomNavigationView mBottomNavigationView;
+	private ArrayList<MenuItem> mMenuItemsArray = new ArrayList<>();
+	// endregion
+
+	public TiUIBottomNavigationTabGroup(TabGroupProxy proxy, TiBaseActivity activity)
+	{
+		super(proxy, activity);
+	}
+
+	@Override
+	public void addViews(TiBaseActivity activity)
+	{
+		// Manually calculate the proper position of the BottomNavigationView.
+		int resourceID = activity.getResources().getIdentifier("design_bottom_navigation_height", "dimen",
+															   activity.getPackageName());
+		this.mBottomNavigationHeightValue = activity.getResources().getDimensionPixelSize(resourceID);
+
+		this.mBottomNavigationView = new BottomNavigationView(activity);
+		setTabMaxWidth();
+
+		// Set the colorPrimary as backgroundColor by default if do not have the backgroundColor set.
+		if (proxy.hasPropertyAndNotNull(TiC.PROPERTY_BACKGROUND_COLOR)) {
+			this.mBottomNavigationView.setBackgroundColor(
+				TiColorHelper.parseColor(proxy.getProperty(TiC.PROPERTY_BACKGROUND_COLOR).toString()));
+		}
```

Comment:
I've figured out how to make the tab bar fill the window. You need to replace the above if-condition with this...
```java
if (proxy.hasPropertyAndNotNull(TiC.PROPERTY_TABS_BACKGROUND_COLOR)) {
	this.mBottomNavigationView.setBackgroundColor(
		TiColorHelper.parseColor(proxy.getProperty(TiC.PROPERTY_TABS_BACKGROUND_COLOR).toString()));
} else {
	try {
		int resourceId = org.appcelerator.titanium.util.TiRHelper.getResource("attr.colorPrimary");
		android.util.TypedValue typedValue = new android.util.TypedValue();
		activity.getTheme().resolveAttribute(resourceId, typedValue, true);
		this.mBottomNavigationView.setBackgroundColor(typedValue.data);
	} catch (Exception ex) {
		Log.e(TAG, "Failed to set BottomNavigationView background color.", ex);
	}
}
```

Note that you need to change `PROPERTY_BACKGROUND_COLOR` to `PROPERTY_TABS_BACKGROUND_COLOR`. The "backgroundProperty" is not supposed to be applied to the tab bar. You need to use "tabsBackgroundColor" property instead. I've verified that this is how it works on iOS too. Our "backgroundProperty" is already correctly applying that color to the `ViewPager`, which is the correct thing to do and matches iOS.

The above works because `BottomNavigationView` is correctly filling the width of the window. It just didn't have a background color applied to it (was transparent by default). So, the trick was to apply the same background color to it as the tabs. I've verified that this works with 1 bottom tab in landscape. It works great. Also, setting the background color this way matches what Google recommends below...
https://material.io/develop/android/components/bottom-navigation-view

The only thing I don't like in my code is how I'm assuming that the default tab color is "colorPrimary". That said, Google does the same in the above link's example, but I wish there was a better way. I tried to do the below in my `else` condition, but it didn't work. Not sure why.
```java
int resourceId = this.mBottomNavigationView.getItemBackgroundResource();
this.mBottomNavigationView.setBackgroundResource(resourceId);
```

Reply:
I just noticed you have a `this.colorPrimaryInt`. Sounds like we should use that in my `else` condition instead since you've already fetched it.